Endless debug logging when server connection closes and retries exhausted

If a server is started (e.g. via StartInsecureControlServer) and a client is connected with a debug logger as in this example:
```
	logger, _ := zap.NewDevelopment()
	_, _ := network.StartControlClient(logging.WithLogger(context.TODO(), logger.Sugar()), &net.Dialer{}, "127.0.0.1:9999")
	select{ }   // wait forever
```
then the server is stopped (either with an interrupt or by calling the server context's cancel function), this appears as the log output:
```
WARN    network/client_connection.go:118    Connection lost, retrying to reconnect 127.0.0.1:9999
WARN    network/client_connection.go:64 Error while trying to connect: dial tcp 127.0.0.1:9999: connect: connection refused
[ ... repeated "clientReconnectionRetry" times ]
WARN    network/client_connection.go:123    Cannot re-dial to target 127.0.0.1:9999: dial tcp 127.0.0.1:9999: connect: connection refused
INFO    network/client_connection.go:105    Closing control client
INFO    network/client_connection.go:107    Connection closed
DEBUG   service/service.go:138  Errors channel closed
DEBUG   service/built_in_handlers.go:35 Error from the connection: %!s(<nil>)
[ ... These two debug lines repeat forever until the client process is forcibly terminated ... ]
```
I presume this is because the infinite loop always selects the "<-c.connection.Errors()" in startPolling() (as evidenced by the output message) and so never reaches the "<-c.ctx.Done()" case which has the only return from the loop.

I'm not completely certain of the intent here or I would offer a solution, but I do know that it should not be spamming the log with these messages.

--- a/pkg/service/service.go
+++ b/pkg/service/service.go
@@ -135,7 +135,10 @@ func (c *service) startPolling() {
 			select {
 			case err, ok := <-c.connection.Errors():
 				if !ok {
-					logging.FromContext(c.ctx).Debugf("Errors channel closed")
+					// The errors channel from a particular connection won't be re-opened once closed
+					// (usually during baseTcpConnection.cleanup), so there's nothing more to be done in this loop.
+					logging.FromContext(c.ctx).Debugf("Errors channel closed, closing polling loop of control service")
+					return
 				}
 				go c.acceptError(err)
 			case <-c.ctx.Done():

